prio_processor/spark/udf.py

```diff
@@ -0,0 +1,253 @@
+import logging
+from collections import Counter
+import array
+
+import pandas as pd
+from prio import PrioContext, libprio
+from prio_processor.prio.commands import import_keys, import_public_keys, match_server
+
+logger = logging.getLogger(__name__)
+
+
+def encode(
+    batch_id: bytes,
+    n_data: int,
+    public_key_hex_internal: bytes,
+    public_key_hex_external: bytes,
+    payload: pd.Series,
+) -> pd.DataFrame:
+    with PrioContext():
```

Comment:
How often is this called? Could it add significant overhead in some cases?

Reply:
This is called once per arrow batch, which by default something like 2-4 GB. It's necessary overhead, but I don't believe it does too much other than initializing the NSS context and setting up the random number generator.